core/src/main/java/com/opensymphony/xwork2/ognl/OgnlUtil.java

```diff
@@ -818,21 +787,25 @@ void internalSetProperty(String name, Object value, Object o, Map<String, Object
         }
     }
 
-    protected Map createDefaultContext(Object root) {
+    protected Map<String, Object> createDefaultContext(Object root) {
         return createDefaultContext(root, null);
     }
 
-    protected Map createDefaultContext(Object root, ClassResolver classResolver) {
+    protected Map<String, Object> createDefaultContext(Object root, ClassResolver classResolver) {
         ClassResolver resolver = classResolver;
         if (resolver == null) {
             resolver = container.getInstance(CompoundRootAccessor.class);
         }
 
         SecurityMemberAccess memberAccess = new SecurityMemberAccess(allowStaticMethodAccess, allowStaticFieldAccess);
-        memberAccess.setExcludedClasses(excludedClasses);
-        memberAccess.setExcludedPackageNamePatterns(excludedPackageNamePatterns);
-        memberAccess.setExcludedPackageNames(excludedPackageNames);
-        memberAccess.setDisallowProxyMemberAccess(disallowProxyMemberAccess);
+        if (devMode) {
```

Comment:
Hello @lukaszlenart.  Allowing `devMode` to disable the exclusion processing makes me a little nervous.

For one thing, _it could produce confusion_ when something works fine locally with `devMode` `true` and then fails due to default exclusions when `devMode` is `false` for a deployed version.  It would also make running `devMode` potentially _riskier_ than it currently is with the exclusion mechanism set/active.

If a developer wants to access classes blocked by the exclusion mechanism (even in `devMode`), they can already modify their local development `struts.xml` configuration to override/set "`struts.excludedClasses`, `struts.excludedPackageNames`, `struts.excludedPackageNamePatterns`" as necessary for that purpose.  Once they are done their testing they can remove the override.

Maybe there could be a compromise in introducing a concept of "`devMode.excludedClasses`, `devMode.excludedPackageNames`, `devMode.excludedPackageNamePatterns`" ?  When in `devMode` those exclusion lists could be used instead of the standard `struts.excludedXXX` ones, with defaults that are slightly more relaxed than the standard ones but still restrictive ?  That would allow some flexibility between modes, still allow for explicit restrictions in devMode (as restrictive or permissive as the developer chooses) and clearly show what is restricted.

What do you think ?

Reply:
Yeah... disabling all the checks isn't the best idea, but the suggestion with `devMode.excludedClasses` and so looks promising, let me add such change, thanks!